examples/logistic/common.py

```diff
@@ -0,0 +1,159 @@
+
+    def __init__(self, num_features):
+        self.w = tfe.define_private_variable(tf.random_uniform([num_features, 1], -0.01, 0.01))
+        self.w_masked = tfe.mask(self.w)
+        self.b = tfe.define_private_variable(tf.zeros([1]))
+        self.b_masked = tfe.mask(self.b)
+
+    @property
+    def weights(self):
+        return self.w, self.b
+
+    def forward(self, x):
+        with tf.name_scope("forward"):
+            out = tfe.matmul(x, self.w_masked) + self.b_masked
+            y = tfe.sigmoid(out)
+            return y
+
+    def backward(self, x, dy, learning_rate=0.01):
+        batch_size = x.shape.as_list()[0]
+        with tf.name_scope("backward"):
+            dw = tfe.matmul(tfe.transpose(x), dy) / batch_size
+            db = tfe.reduce_sum(dy, axis=0) / batch_size
+            assign_ops = [
+                tfe.assign(self.w, self.w - dw * learning_rate),
+                tfe.assign(self.b, self.b - db * learning_rate),
+            ]
+            return assign_ops
+
+    def loss_grad(self, y, y_hat):
+        with tf.name_scope("loss-grad"):
+            dy = y_hat - y
+            return dy
+
+    def fit_batch(self, x, y):
+        with tf.name_scope("fit-batch"):
+            y_hat = self.forward(x)
+            dy = self.loss_grad(y, y_hat)
+            fit_batch_op = self.backward(x, dy)
+            return fit_batch_op
+
+    def fit(self, sess, x, y, num_batches):
+        fit_batch_op = self.fit_batch(x, y)
+        for batch in range(num_batches):
+            print("Batch {0: >4d}".format(batch))
+            sess.run(fit_batch_op, tag='fit-batch')
+
+    def evaluate(self, sess, x, y, data_owner):
+
+        def print_accuracy(y_hat, y) -> tf.Operation:
+            with tf.name_scope("print-accuracy"):
+                correct_prediction = tf.equal(tf.round(y_hat), y)
+                accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+                print_op = tf.print("Accuracy on {}:".format(data_owner.player_name), accuracy)
+                return print_op
+
+        with tf.name_scope("evaluate"):
+            y_hat = self.forward(x)
+            print_accuracy_op = tfe.define_output(data_owner.player_name, [y_hat, y], print_accuracy)
+
+        sess.run(print_accuracy_op, tag='evaluate')
+
+
+class DataOwner:
+
+    def __init__(self, player_name, num_features, training_set_size, test_set_size, batch_size):
+        self.player_name = player_name
+        self.num_features = num_features
+        self.training_set_size = training_set_size
+        self.test_set_size = test_set_size
+        self.batch_size = batch_size
+        self.train_initializer = None
+        self.test_initializer = None
+
+    @property
+    def initializer(self):
+        return tf.group(self.train_initializer, self.test_initializer)
+
+    def provide_training_data(self):
+
+        def norm(x, y):
+            return tf.cast(x, tf.float32), tf.expand_dims(y, 0)
+
+        x_raw = tf.random.uniform(
```

Comment:
how much work would it be to replace this with real data? if not this PR, maybe open an issue for it

Reply:
all for this if/when we find the right use case!